src/object/srv_obj.c

```diff
@@ -1314,37 +1324,24 @@ ds_obj_query_key_handler(crt_rpc_t *rpc)
 }
 
 /**
- * Choose abt pools for object RPC. Because dkey enumeration might create ULT
- * on other xstream pools, so we have to put it to the shared pool. For other
- * RPC, it can be put to the private pool. XXX we might just instruct the
- * server create task for inline I/O.
+ * Choose abt pools for object RPC. Because those update RPC might create pool
+ * map refresh ULT, let's put it to share pool. For other RPC, it can be put to
+ * the private pool. XXX we might just instruct the server create task for
+ * inline I/O.
  */
 ABT_pool
 ds_obj_abt_pool_choose_cb(crt_rpc_t *rpc, ABT_pool *pools)
 {
-	struct obj_rw_in	*orw;
-	struct obj_punch_in	*opi;
-	ABT_pool		 pool;
+	ABT_pool	pool;
 
 	switch (opc_get(rpc->cr_opc)) {
 	case DAOS_OBJ_RPC_ENUMERATE:
-	case DAOS_OBJ_DKEY_RPC_ENUMERATE:
 	case DAOS_OBJ_RPC_PUNCH:
-		pool = pools[DSS_POOL_SHARE];
-		break;
 	case DAOS_OBJ_RPC_PUNCH_DKEYS:
 	case DAOS_OBJ_RPC_PUNCH_AKEYS:
-		/* if the update/punch need to dispatch to other tgts, schedules
-		 * it in SHARE pool as need other ES to dispatch it.
-		 */
-		opi = crt_req_get(rpc);
-		pool = (opi->opi_shard_tgts.ca_arrays != NULL) ?
-		       pools[DSS_POOL_SHARE] : pools[DSS_POOL_PRIV];
-		break;
 	case DAOS_OBJ_RPC_UPDATE:
-		orw = crt_req_get(rpc);
-		pool = (orw->orw_shard_tgts.ca_arrays != NULL) ?
-		       pools[DSS_POOL_SHARE] : pools[DSS_POOL_PRIV];
+	case DAOS_OBJ_RPC_FETCH:
+		pool = pools[DSS_POOL_SHARE];
```

Comment:
since the pool map ULT is async and need not ABT wait it, why still need to change it to be always SHARE POOL?

Reply:
Yes, with pool map async update ULT, all object rpc might be SHARE POOL.